woocommerce/woocommerce-ios
# [Store onboarding] Upgrade plan from launch store screen

Closes: #9141


## Description

For free trial WPCOM sites, we cannot publish the site without upgrading the plan.

This PR,
- Adds a banner in the "Launch store" screen
- Enables the "Publish My Store" button only if the site is not under free trial

## Testing instructions

Prerequisite: Create a free trial by visiting - https://wordpress.com/setup/wooexpress/

- Login into the app using the free trial site
- In the dashboard screen, you should also see the "X days left in your trial.." banner at the bottom of the screen.
- You should also see the "Launch your store" task on the dashboard screen. 
- Tap on the "Launch your store" task. 
- In the "Preview" screen, verify that you see a banner at the top asking you to upgrade the plan to launch your store
- Also validate the "Publish My Store" button is disabled
- Tap on the `Upgrade` button and purchase a plan (using credits)
- You should be navigated back to the dashboard after purchasing a plan.
- Validate that you no longer see the "X days left in your trial.." banner at the bottom of the dashboard screen
- If you don't see "Launch your store" task, pull to refresh to reload the onboarding task list. (Takes a few secs for the purchase to take effect)
- Tap on "Launch your store" task, verify that now the "Publish My Store" button is enabled, and the "Upgrade plan" banner is no longer visible.

## Screenshots

Launch store screen
| Site | Dark | Light |
|--------|--------|--------|
| Free trial |  ![Dark](https://user-images.githubusercontent.com/524475/228572757-085373c4-5d31-4266-a449-e1cf0f098fd1.png) | ![Light](https://user-images.githubusercontent.com/524475/228572798-9b34457e-bb67-49f9-b517-29cdd28ef626.png) |
| WPCOM after upgrade plan | ![D-ReadyToPublish](https://user-images.githubusercontent.com/524475/228532578-cacf81c2-5fbe-4bb2-a14b-b9e52a4f79b0.png) | ![L-ReadyToPublish](https://user-images.githubusercontent.com/524475/228532594-715aee82-0d0e-4ad6-91d5-1368581dc14d.png) |


---
